### PR TITLE
story/TYTA-99/securize-edit-and-delete-scope-of-users 

### PR DESCRIPTION
**Pull request by remirubis**
### Description

How I secure this routes : 
- I check if the user id in the request sent by the auth middleware equals the user id provided in the request
- I check in projects or teams if the user who tries to edit data have admin role

**Checklist of actions**:
- [x] Securitize project
- [x] Securitize user
- [x] Securitize tasks
- [x] Securitize teams
- [ ] Securitize time tasks

### ⚠️ CAUTION ⚠️ Merge [TYTA-98](https://github.com/TimeYourTask/Tyta-cors/pull/41) before this PR

[TYTA-98]: https://timeyourtask.atlassian.net/browse/TYTA-98?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/controllers/teamCtrl.js [hunk 1]
```diff
@@ -216,10 +263,24 @@ exports.getUserOfTeam = (req, res) => {
   Team.findById(req.params.teamId)
     .populate(['users.user'])
     .then((team) => {
+      if (team) {
```
Comment (FabienHP) [inline src/controllers/teamCtrl.js]:
```suggestion
      if (!team) {
```
Its better for a not found 😂

File: src/controllers/teamCtrl.js [hunk 2]
```diff
@@ -216,10 +263,24 @@ exports.getUserOfTeam = (req, res) => {
...
+        return res
+          .status(404)
+          .json({ message: 'No user found for this team!' });
+      }
+
+      // Blocks users who want to access other users without being admin
+      const isUserInTeam = team.users.some(
+        (user) => user.user.toString() === req.userId
```
Comment (FabienHP) [inline src/controllers/teamCtrl.js]:
```suggestion
        (user) => user.user.id.toString() === req.userId
```
'user.id' you mean ?

**Review by FabienHP — CHANGES_REQUESTED** [review]

File: src/controllers/teamCtrl.js (hunk 1 above)
Comment (remirubis) [reply src/controllers/teamCtrl.js]: > ```suggestion
> 
>       if (!team) {
> 
> ```
> 
> Its better for a not found 😂

Good catch! Thx!

File: src/controllers/teamCtrl.js (hunk 2 above)
Comment (remirubis) [reply src/controllers/teamCtrl.js]: Yes it should be with 'id' but I have to test it before make a change, thanks for the catch !

Comment (remirubis) [reply src/controllers/teamCtrl.js]: Nice catch! It's write ✅

**Pull request merged**